patroni/api.py

```diff
@@ -532,8 +533,36 @@ def __get_ssl_options(config):
 
     def __set_config_parameters(self, config):
         self.__auth_key = base64.b64encode(config['auth'].encode('utf-8')).decode('utf-8') if 'auth' in config else None
-        self.connection_string = '{0}://{1}/patroni'.format(self.__protocol,
-                                                            config.get('connect_address') or self.__listen)
+        self.connection_string = uri(self.__protocol, config.get('connect_address') or self.__listen, 'patroni')
+
+    @staticmethod
+    def __has_dual_stack():
+        if hasattr(socket, 'AF_INET6') and hasattr(socket, 'IPPROTO_IPV6') and hasattr(socket, 'IPV6_V6ONLY'):
+            sock = None
+            try:
+                sock = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)
+                sock.setsockopt(socket.IPPROTO_IPV6, socket.IPV6_V6ONLY, False)
+                return True
+            except socket.error as e:
+                logger.debug('Error when working with ipv6 socket: %s', e)
+            finally:
+                if sock:
+                    sock.close()
+        return False
+
+    def __httpserver_init(self, host, port):
+        dual_stack = self.__has_dual_stack()
+        if host == '':
+            host = None
+        if host is None and dual_stack:
+            host = '::'
```

Comment:
Is this required? Python docs says, that if

```
By passing None as the value of host and port, you can
pass NULL to the underlying C API.
```

and in this case:

```
the network address will be set to the loopback interface address
(INADDR_LOOPBACK for IPv4 addresses, IN6ADDR_LOOPBACK_INIT
for IPv6 address)
```

So looks like with `None` result would be the same.

Reply:
You are right. Will remove it.